src/libelf-integers.js

```diff
@@ -23,17 +23,35 @@ var ElfUInt = function (width) {
         } else {
             // Initialize from Number
             if (typeof value === 'number') {
+                if (!Number.isSafeInteger(value)) {
+                    console.warn(
+                        'Libelf.js: number ' + value +
+                        ' is beyond 53 bits integer precision, use other initialization formats for better precision'
+                    );
+                }
+                if (value > Math.pow(2, this.width)-1 || value < -Math.pow(2, this.width)) {
+                    console.warn(
+                        'Libelf.js: number ' + value +
+                        ' overflows ' + this.width + ' bits width, use larger width to keep higher bits'
+                    );
+                }
                 for (var i = 0; i < this.chunks.length; i++) {
-                    this.chunks[i] = value & 0xFFFF;
-                    value >>>= 16;
+                    // Apply mask on the last chunk to cast to correct width
+                    this.chunks[i] = value & (i === this.chunks.length - 1 ? this.mask : 0xFFFF);
+                    if (value < 0) {
+                        // Fix round off of bits in negative values
+                        value -= 65535
+                    }
+                    value /= 65536;
                 }
             }
             // Initialize from String
             if (typeof value === 'string') {
                 if (value.toLowerCase().startsWith("0x"))
                     value = value.slice(2);
                 for (var i = 0; i < this.chunks.length; i++) {
-                    this.chunks[i] = parseInt(value.slice(-4), 16);
+                    // Force NaN value to be 0
+                    this.chunks[i] = parseInt(value.slice(-4), 16) || 0;
```

Comment:
nit: Even if the end-result is the same, can you replace `|| 0` with `|0` (change logical OR into bitwise OR)?
The latter is the de-facto standard to "integerize" JavaScript numbers, cf. http://asmjs.org/spec/latest/